Yank/yamlbuild.py

```diff
@@ -1177,14 +1178,17 @@ class YamlBuilder:
     def yank_options(self):
         return self._isolate_yank_options(self.options)
 
-    def __init__(self, yaml_source=None):
+    def __init__(self, yaml_source=None, platform_name=None):
```

Comment:
There's not a great solution to this, but using `Platform.getSpeed()` sounds like the best approach for now. We could later add checks to see if double precision is required/supported, but it may be best to work to improve the OpenMM `Platform` API instead so we don't have to do this within YANK.